client/allocrunner/taskrunner/sids_hook_test.go

```diff
@@ -297,11 +296,7 @@ func TestTaskRunner_DeriveSIToken_UnWritableTokenFile(t *testing.T) {
 	go tr.Run()
 
 	// wait for task runner to finish running
-	select {
-	case <-tr.WaitCh():
-	case <-time.After(time.Duration(testutil.TestMultiplier()*15) * time.Second):
-		r.Fail("timed out waiting for task runner")
-	}
+	testWaitForTaskToDie(t, tr)
```

Comment:
Changes like this had to be done because the task runner `Run()` method doesn't return anymore, it gets blocked waiting for a possible restart, so `WaitCh` is not closed until the task reaches a point where it will not restart anymore.